cdap-integration-test/src/main/java/co/cask/cdap/test/remote/RemoteStreamManager.java

```diff
@@ -0,0 +1,133 @@
+ * the License at
+ *  
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.test.remote;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.flow.flowlet.StreamEvent;
+import co.cask.cdap.client.StreamClient;
+import co.cask.cdap.client.config.ClientConfig;
+import co.cask.cdap.common.exception.StreamNotFoundException;
+import co.cask.cdap.test.StreamManager;
+import com.google.common.base.Charsets;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Lists;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link StreamManager} that interacts with stream service via HTTP calls
+ */
+public class RemoteStreamManager implements StreamManager {
+  private final StreamClient streamClient;
+  private final String streamName;
+
+  public RemoteStreamManager(ClientConfig clientConfig, String streamName) {
+    this.streamClient = new StreamClient(clientConfig);
+    this.streamName = streamName;
+  }
+
+  @Override
+  public void createStream() throws IOException {
+    try {
+      streamClient.create(streamName);
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void send(String content) throws IOException {
+    try {
+      streamClient.sendEvent(streamName, content);
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void send(byte[] content) throws IOException {
+    try {
+      streamClient.sendEvent(streamName, new String(content, Charsets.UTF_8));
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void send(byte[] content, int off, int len) throws IOException {
+    try {
+      streamClient.sendEvent(streamName, new String(content, off, len, Charsets.UTF_8));
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void send(ByteBuffer buffer) throws IOException {
+    try {
+      streamClient.sendEvent(streamName, Bytes.toString(buffer));
+    } catch (IOException e) {
+      throw e;
+    } catch (Exception e) {
+      throw Throwables.propagate(e);
+    }
+  }
+
+  @Override
+  public void send(Map<String, String> headers, String content) throws IOException {
+    throw new UnsupportedOperationException("TODO");
```

Comment:
Not currently, doesn't look like ``StreamClient`` allows you to pass in headers. Created [CDAP-2211](https://issues.cask.co/browse/CDAP-2211)